vividus-plugin-saucelabs/src/main/java/org/vividus/saucelabs/SauceLabsTestLinkPublisher.java

```diff
@@ -18,27 +18,36 @@
 
 import static java.lang.String.format;
 
+import java.util.Map;
+import java.util.Optional;
+
 import com.google.common.eventbus.EventBus;
 import com.saucelabs.saucerest.DataCenter;
 
 import org.vividus.selenium.IWebDriverProvider;
 import org.vividus.selenium.cloud.AbstractCloudTestLinkPublisher;
+import org.vividus.selenium.manager.GenericWebDriverManager;
 import org.vividus.testcontext.TestContext;
 
 public class SauceLabsTestLinkPublisher extends AbstractCloudTestLinkPublisher
 {
     private final String sauceLabsUrl;
+    private final GenericWebDriverManager genericWebDriverManager;
 
     public SauceLabsTestLinkPublisher(DataCenter dataCenter, IWebDriverProvider webDriverProvider, EventBus eventBus,
-            TestContext testContext)
+            TestContext testContext, GenericWebDriverManager genericWebDriverManager)
     {
         super("SauceLabs", webDriverProvider, eventBus, testContext);
         this.sauceLabsUrl = dataCenter.appServer();
+        this.genericWebDriverManager = genericWebDriverManager;
     }
 
     @Override
     protected String getCloudTestUrl(String sessionId) throws GetCloudTestUrlException
     {
-        return format("%stests/%s", sauceLabsUrl, sessionId);
+        Map<String, Object> capabilities = genericWebDriverManager.getCapabilities().asMap();
+        return Optional.ofNullable(capabilities.get("appium:testobject_test_report_url"))
```

Comment:
```suggestion
        return Optional.ofNullable(genericWebDriverManager.getCapabilities().getCapability("appium:testobject_test_report_url"))
```

Reply:
it looks too long, can we at least make Capabilities capabilities = genericWebDriverManager.getCapabilities() as a separate statement?

Reply:
up to you, my point was to avoid conversion to Map